sdks/python/apache_beam/yaml/main.py

```diff
@@ -67,6 +105,12 @@ def _pipeline_spec_from_args(known_args):
 def run(argv=None):
   known_args, pipeline_args = _configure_parser(argv)
   pipeline_yaml = _pipeline_spec_from_args(known_args)
+  jinja_variables = _extract_jinja_variables(known_args, pipeline_args)
+  if (known_args.jinja_variables is not None or
+      known_args.flags_as_jinja_variables):
+    pipeline_yaml = jinja2.Environment(
+        undefined=jinja2.StrictUndefined).from_string(pipeline_yaml).render(
```

Comment:
Doing something like this:

```suggestion
    class FileIOLoader(jinja2.BaseLoader):
      def get_source(self, environment, path):
        source = FileSystems.open(path).read().decode()
        return source, path, lambda: True
          
    pipeline_yaml = jinja2.Environment(
        undefined=jinja2.StrictUndefined, 
        loader=FileIOLoader()).from_string(pipeline_yaml).render(
```

Allows one to specify something like
```
pipeline:
  transforms:
    ...
providers:
{% include providers.yaml %}
```

Where the providers could be defined as a file locally or in GCS. This also works for any instance of 
`{% include some_file %}` within the template

Reply:
Then I had 
```
pipeline_yaml = os.linesep.join(
      [s for s in pipeline_yaml.splitlines() if s.strip()])
```
for formatting after rendering to remove the empty lines so that YAML is prettier in display_data

Reply:
Good idea.